### PR TITLE
[TK-05023] Integration waits for holding deps

### PR DESCRIPTION
**Pull request by freesig**
Puts the holding checks in integration so they can be removed from validation in an upcoming PR

### CODE REVIEW

**Review by neonphog — APPROVED** [review]

File: crates/holochain/src/core/workflow/integrate_dht_ops_workflow.rs [hunk 1]
```diff
@@ -176,107 +160,130 @@ pub async fn integrate_dht_ops_workflow(
...
+                    {
+                        Some(prev_header) => {
+                            let op_hash = DhtOpHash::with_data_sync(
+                                &UniqueForm::RegisterAgentActivity(prev_header.header()),
+                            );
+                            if workspace.integration_limbo.contains(&op_hash)? {
+                                return Ok(true);
+                            }
+                        }
+                        None => return Ok(false),
```
Comment (maackle) [inline crates/holochain/src/core/workflow/integrate_dht_ops_workflow.rs:247]: Just a style thing: early returns make me wonder what other branch of logic I might be missing, and this is a case where they could probably be removed to let the whole function return all at once. But it's such a short function (nice!) that it doesn't make that much difference in this case.

File: crates/holochain/src/core/workflow/integrate_dht_ops_workflow.rs [hunk 2]
```diff
@@ -286,6 +293,52 @@ fn op_dependencies_held<P: PrefixType>(
     }
 }
 
+/// Get an element that contains an entry from the cascade
```
Comment (maackle) [inline crates/holochain/src/core/workflow/integrate_dht_ops_workflow.rs:296]:
```suggestion
/// Check the cascade to see if this Header is also stored with an Entry
```

File: crates/holochain/src/core/workflow/integrate_dht_ops_workflow.rs [hunk 3]
```diff
@@ -286,6 +293,52 @@ fn op_dependencies_held<P: PrefixType>(
...
+            ElementEntry::NotApplicable => Err(DhtOpConvertError::HeaderEntryMismatch.into()),
+            ElementEntry::NotStored => {
+                Err(DhtOpConvertError::MissingEntryDataForHeader(hash.clone()).into())
+            }
+        },
+        None => Ok(false),
+    }
+}
+
+/// Get an entry that contains an entry from the cascade
```
Comment (maackle) [inline crates/holochain/src/core/workflow/integrate_dht_ops_workflow.rs:318]:
```suggestion
/// Check if an Entry is stored in the cascade
```

File: crates/holochain/src/core/workflow/integrate_dht_ops_workflow.rs [hunk 4]
```diff
@@ -286,6 +293,52 @@ fn op_dependencies_held<P: PrefixType>(
...
+    match cascade
+        .retrieve_entry(hash.clone(), Default::default())
+        .await?
+    {
+        Some(_) => Ok(true),
+        None => Ok(false),
+    }
+}
+
+/// Get an header that contains an entry from the cascade
```
Comment (maackle) [inline crates/holochain/src/core/workflow/integrate_dht_ops_workflow.rs:331]:
```suggestion
/// Check if a header is stored in the cascade

**Review by maackle — APPROVED** [review]
Things are looking much clearer! Happy with how the simplification is going.

**Comment by freesig** [comment]
Seems to be something going on with the sys validation test. I'll hold off on merging this until I fix that.

**Pull request merged**